### ISSUE TITLE
Fails with:

### ISSUE THREAD

**Issue by georgij** [Adyen/adyen-android#22]
Unable to resolve dependency for ':customwithcheckoutui@debug/compileClasspath': Could not resolve project :adyen-core.
Open File
Show Details

Unable to resolve dependency for ':customwithcheckoutui@debug/compileClasspath': Could not resolve project :adyen-ui.
Open File
Show Details

Unable to resolve dependency for ':customwithcheckoutui@debugAndroidTest/compileClasspath': Could not resolve project :adyen-core.
Open File
Show Details

Unable to resolve dependency for ':customwithcheckoutui@debugAndroidTest/compileClasspath': Could not resolve project :adyen-ui.
Open File
Show Details

Unable to resolve dependency for ':customwithcheckoutui@debugUnitTest/compileClasspath': Could not resolve project :adyen-core.
Open File
Show Details

Unable to resolve dependency for ':customwithcheckoutui@debugUnitTest/compileClasspath': Could not resolve project :adyen-ui.
Open File
Show Details

Unable to resolve dependency for ':customwithcheckoutui@release/compileClasspath': Could not resolve project :adyen-core.
Open File
Show Details

Unable to resolve dependency for ':customwithcheckoutui@release/compileClasspath': Could not resolve project :adyen-ui.
Open File
Show Details

Unable to resolve dependency for ':customwithcheckoutui@releaseUnitTest/compileClasspath': Could not resolve project :adyen-core.
Open File
Show Details

Unable to resolve dependency for ':customwithcheckoutui@releaseUnitTest/compileClasspath': Could not resolve project :adyen-ui.
Open File
Show Details

Unable to resolve dependency for ':adyen-ui@debug/compileClasspath': Could not resolve project :adyen-core.
Open File
Show Details

Unable to resolve dependency for ':adyen-ui@debug/compileClasspath': Could not resolve project :adyen-utils.
Open File
Show Details

Unable to resolve dependency for ':adyen-ui@debug/compileClasspath': Could not resolve project :adyen-cardscan.
Open File
Show Details

Unable to resolve dependency for ':adyen-ui@debugAndroidTest/compileClasspath': Could not resolve project :adyen-core.
Open File
Show Details

Unable to resolve dependency for ':adyen-ui@debugAndroidTest/compileClasspath': Could not resolve project :adyen-utils.
Open File
Show Details

Unable to resolve dependency for ':adyen-ui@debugAndroidTest/compileClasspath': Could not resolve project :adyen-cardscan.
Open File
Show Details

Unable to resolve dependency for ':adyen-ui@debugUnitTest/compileClasspath': Could not resolve project :adyen-core.
Open File
Show Details

Unable to resolve dependency for ':adyen-ui@debugUnitTest/compileClasspath': Could not resolve project :adyen-utils.
Open File
Show Details

Unable to resolve dependency for ':adyen-ui@debugUnitTest/compileClasspath': Could not resolve project :adyen-cardscan.
Open File
Show Details

Unable to resolve dependency for ':adyen-ui@release/compileClasspath': Could not resolve project :adyen-core.
Open File
Show Details

Unable to resolve dependency for ':adyen-ui@release/compileClasspath': Could not resolve project :adyen-utils.
Open File
Show Details

Unable to resolve dependency for ':adyen-ui@release/compileClasspath': Could not resolve project :adyen-cardscan.
Open File
Show Details

Unable to resolve dependency for ':adyen-ui@releaseUnitTest/compileClasspath': Could not resolve project :adyen-core.
Open File
Show Details

Unable to resolve dependency for ':adyen-ui@releaseUnitTest/compileClasspath': Could not resolve project :adyen-utils.
Open File
Show Details

Unable to resolve dependency for ':adyen-ui@releaseUnitTest/compileClasspath': Could not resolve project :adyen-cardscan.
Open File
Show Details

Unable to resolve dependency for ':adyen-androidpay@debug/compileClasspath': Could not resolve project :adyen-core.
Open File
Show Details

Unable to resolve dependency for ':adyen-androidpay@debugAndroidTest/compileClasspath': Could not resolve project :adyen-core.
Open File
Show Details

Unable to resolve dependency for ':adyen-androidpay@debugUnitTest/compileClasspath': Could not resolve project :adyen-core.
Open File
Show Details

Unable to resolve dependency for ':adyen-androidpay@release/compileClasspath': Could not resolve project :adyen-core.
Open File
Show Details

Unable to resolve dependency for ':adyen-androidpay@releaseUnitTest/compileClasspath': Could not resolve project :adyen-core.
Open File
Show Details

Unable to resolve dependency for ':customuiapplication@debug/compileClasspath': Could not resolve project :adyen-core.
Open File
Show Details

Unable to resolve dependency for ':customuiapplication@debugAndroidTest/compileClasspath': Could not resolve project :adyen-core.
Open File
Show Details

Unable to resolve dependency for ':customuiapplication@debugUnitTest/compileClasspath': Could not resolve project :adyen-core.
Open File
Show Details

Unable to resolve dependency for ':customuiapplication@release/compileClasspath': Could not resolve project :adyen-core.
Open File
Show Details

Unable to resolve dependency for ':customuiapplication@releaseUnitTest/compileClasspath': Could not resolve project :adyen-core.
Open File
Show Details

Unable to resolve dependency for ':defaultApp@debug/compileClasspath': Could not resolve project :adyen-core.
Open File
Show Details

Unable to resolve dependency for ':defaultApp@debug/compileClasspath': Could not resolve project :adyen-androidpay.
Open File
Show Details

Unable to resolve dependency for ':defaultApp@debug/compileClasspath': Could not resolve project :adyen-ui.
Open File
Show Details

Unable to resolve dependency for ':defaultApp@debugAndroidTest/compileClasspath': Could not resolve project :adyen-core.
Open File
Show Details

Unable to resolve dependency for ':defaultApp@debugAndroidTest/compileClasspath': Could not resolve project :adyen-androidpay.
Open File
Show Details

Unable to resolve dependency for ':defaultApp@debugAndroidTest/compileClasspath': Could not resolve project :adyen-ui.
Open File
Show Details

Unable to resolve dependency for ':defaultApp@debugUnitTest/compileClasspath': Could not resolve project :adyen-core.
Open File
Show Details

Unable to resolve dependency for ':defaultApp@debugUnitTest/compileClasspath': Could not resolve project :adyen-androidpay.
Open File
Show Details

Unable to resolve dependency for ':defaultApp@debugUnitTest/compileClasspath': Could not resolve project :adyen-ui.
Open File
Show Details

Unable to resolve dependency for ':defaultApp@release/compileClasspath': Could not resolve project :adyen-core.
Open File
Show Details

Unable to resolve dependency for ':defaultApp@release/compileClasspath': Could not resolve project :adyen-androidpay.
Open File
Show Details

Unable to resolve dependency for ':defaultApp@release/compileClasspath': Could not resolve project :adyen-ui.
Open File
Show Details

Unable to resolve dependency for ':defaultApp@releaseUnitTest/compileClasspath': Could not resolve project :adyen-core.
Open File
Show Details

Unable to resolve dependency for ':defaultApp@releaseUnitTest/compileClasspath': Could not resolve project :adyen-androidpay.
Open File
Show Details

Unable to resolve dependency for ':defaultApp@releaseUnitTest/compileClasspath': Could not resolve project :adyen-ui.
Open File
Show Details

**Comment by timon-langlotz** [comment]
Hi @georgij,

I just did a fresh clone from Github and imported the project into AndroidStudio by selecting `File > Open...` and then selecting the `build.gradle` file in the root of the project, and that worked.

I'm assuming that AndroidStudio has updated your `build.gradle` file or the `gradle-wrapper.properties` file.

Could you check with Git if these files (or any other files) have any uncommitted changes?
And could you also make sure that you have the required build tools downloaded? Usually AndroidStudio prompts you to update when there's software missing or updates available.

Cheers,
Timon

**Comment by georgij** [comment]
Yes, there are some version updates, like Grade 4.4 instead of 3.3 but if I roll it back it would not even try to build.

**Comment by timon-langlotz** [comment]
How does your `build.gradle (adyen-android)` look like? Does it say:
```
buildscript {
    repositories {
        jcenter()
    }
    dependencies {
        classpath 'com.android.tools.build:gradle:2.3.3'
    }
}
```
?

What errors are you getting when trying to build with the unchanged gradle versions? And how do you build? From the IDE or from the command line?

Cheers,
Timon

**Comment by georgij** [comment]
I have it updated to:
`buildscript {
    repositories {
        jcenter()
        maven {
            url 'https://maven.google.com/'
            name 'Google'
        }
    }
    dependencies {
        classpath 'com.android.tools.build:gradle:3.1.0'
    }
}
`
If I set it back I get: "... we strongly recommend that you update the Android Gradle plugin to version 3.1.0 and Gradle to version 4.4."

**Comment by georgij** [comment]
Ah, it looks like I can have gradle set to 4.4 and the gradle plugin left with 2.3.3 and it works.

**Comment by timon-langlotz** [comment]
Thanks for the details @georgij!

Unfortunately, we cannot update to this Gradle version for now, since this would result in breaking changes (hence breaking backwards compatability). You can find more details about these breaking changes here: https://developer.android.com/studio/build/gradle-plugin-3-0-0-migration.html

There are however some rumors that we will provide an Android SDK in the near future with lots of updates and improvements (including support for the latest version of Gradle, Material design, etc.)! 👌 

Stay tuned!
Timon

**Comment by timon-langlotz** [comment]
Hi @georgij,

we've just released version 2.0.0, and that one uses the latest gradle version.

Cheers,
Timon
